### PR TITLE
[MISC] Alphabet: Move some views to alphabet views - Part 3

### PR DESCRIPTION
**Pull request by Irallia**
Third and last part

Resolves https://github.com/seqan/product_backlog/issues/337 and
resolves https://github.com/seqan/product_backlog/issues/338 and
resolves https://github.com/seqan/product_backlog/issues/339 and
resolves https://github.com/seqan/product_backlog/issues/340 and
resolves https://github.com/seqan/product_backlog/issues/341 and
resolves https://github.com/seqan/product_backlog/issues/342.

Moves snippets and tests.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -128,6 +128,9 @@ regression test suite and patches at https://github.com/seqan/seqan3/tree/master
 * Renamed `seqan3::phred68legacy` to `seqan3::phred68solexa` ([\#2522](https://github.com/seqan/seqan3/pull/2522)).
 * Renamed `seqan3::sam_dna16` to `seqan3::dna16sam`
   ([\#2521](https://github.com/seqan/seqan3/pull/2521)).
+* Moved some views from `seqan3/range/views/` to `seqan3/alphabet/views/`: `seqan3::char_to`, `seqan3::to_char`,
+  `seqan3::to_rank`, `seqan3::translate_join`, `seqan3::translate`, `seqan3::trim_quality`
+  ([\#2530](https://github.com/seqan/seqan3/pull/2530)) && ([\#...](https://github.com/seqan/seqan3/pull/...)).
```
Comment (marehr) [inline CHANGELOG.md:133]:
```suggestion
```

Comment (marehr) [reply CHANGELOG.md:133]: (we list all moved header files before the release)

File: include/seqan3/alphabet/views/translate_join.hpp [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+ * the number of translation frames given. Therefore, it has the same capabilities as the standard view_translate but concatenates the different
+ * frames of the different input sequences rather than having a separate range for each input sequence containing the translated frames. In the output,
+ * frames are ordered in a way, that all requested frames are listed per sequence directly after each other in the order of input sequences.
+ * improved and efficient downstream post-processing if needed. However, the index of a frame for a specific sequence needs to be calculated via
+ * modulo operations in this case. The i-th frame of the j-th sequence can be calculated by n = (i * s) + j, where s is the number of frames used
+ * for translation (index starting at zero).
+ *
+ * In short, this views behaves the same as:
+ *
+ * TODO: make this a snippet
```
Comment (Irallia) [inline include/seqan3/alphabet/views/translate_join.hpp:351]: Is the snippet in the later example not enough?
 -> \snippet test/snippet/alphabet/views/translate_join.cpp example

File: include/seqan3/alphabet/views/translate_join.hpp [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+ * ```cpp
+ * std::vector<std::vector<dna5>> vec {...};
+ * auto v = vec | views::translate | views::join;
+ * ```
```
Comment (Irallia) [inline include/seqan3/alphabet/views/translate_join.hpp:355]: Is the snippet in the later example not enough and this remains a note?
 -> \snippet test/snippet/alphabet/views/translate_join.cpp example

**Review by MitraDarja — APPROVED** [review]

File: include/seqan3/alphabet/views/all.hpp [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Meta-header for the \link views views submodule \endlink.
```
Comment (eseiler) [inline include/seqan3/alphabet/views/all.hpp:9]:
```suggestion
 * \brief Meta-header for alphabet-related views.
```

File: test/performance/alphabet/CMakeLists.txt [hunk 5]
```diff
@@ -2,3 +2,6 @@ seqan3_benchmark(alphabet_assign_char_benchmark.cpp)
 seqan3_benchmark(alphabet_assign_rank_benchmark.cpp)
 seqan3_benchmark(alphabet_to_char_benchmark.cpp)
 seqan3_benchmark(alphabet_to_rank_benchmark.cpp)
+seqan3_benchmark(view_translate_1D_benchmark.cpp)
```
Comment (eseiler) [inline test/performance/alphabet/CMakeLists.txt:5]: Can you put these into `test/performance/alphabet/views/`

**Review by eseiler — CHANGES_REQUESTED** [review]

**Review by eseiler — APPROVED** [review]

**Comment by eseiler** [comment]
Needs a rebase

**Pull request merged**